sample_information_validator.py

```diff
@@ -23,78 +23,280 @@
-            return True
-        except ValueError:
+            'Project not found, please check your entry for the PlateID, it should have the format'
+            'PxxxxxPx, where x are numbers. If your Plate ID is correct, contact your project coordinator.'
+            )
+            quit()
+        # more than one project found
+        elif len(prow) > 1:
             logger.error(
-                    'Cell {} is numeric but cannot be '
-                    'transformed to float'.format(cell.coordinate)
+            'Project ID not unique, please check your entry for the PlateID, it should have the format'
+            'PxxxxxPx, where x are numbers. If your Plate ID is correct, contact your project coordinator.'
             )
-            return False
-        except TypeError:
-            if cell.value is None:
-                logger.warning(
-                    'Cell {} is numeric but empty'.format(cell.coordinate)
+            quit()
+        else:
+        # puts the Document of the identified project in a new variable "pdoc"
+            pdoc = db.get(prow[0].id)
+            return(pdoc)
+
+    def requirements(self, path):
+        if self.sample_rec == None:
+            wb_sample_recommendations = load_workbook(path, read_only=True, data_only=True)
+            self.sample_rec = wb_sample_recommendations['Numeric']
+
+    def prep_standards(self, info, recom_path):
+        ''' gets the sample requirements from the sample requirement excel sheet based
+            on the given sample prep type. '''
+
+        prep = info['details']['library_construction_method']
+
+        prep_recs = [None,None,None,None,None,None,None,None]
+        prep_type_found = False
+        for row in range(2, 15):
+            cellID_prep = "A{row}".format(row=row)
+            self.requirements(recom_path)
+            if(self.sample_rec[cellID_prep].value == prep):
+                prep_row = row
+                prep_recs = [\
+                self.sample_rec["C{min_conc}".format(min_conc=prep_row)].value, \
+                self.sample_rec["D{max_conc}".format(max_conc=prep_row)].value, \
+                self.sample_rec["E{min_vol}".format(min_vol=prep_row)].value, \
+                self.sample_rec["F{rec_ng}".format(rec_ng=prep_row)].value, \
+                self.sample_rec["G{min_ng}".format(min_ng=prep_row)].value, \
+                self.sample_rec["H{qual_req}".format(qual_req=prep_row)].value, \
+                self.sample_rec["I{qual_rec}".format(qual_rec=prep_row)].value]
+                prep_type_found = True
+        if(prep_type_found == False):
+            logger.error(
+                'Preparation type \"{}\" not found'.format(prep)
+                )
+            quit()
+        return(prep_recs)
+
+    def validate_project_Name(self, info):
+        project_name_DB = info['project_name']
+        project_name_user = re.split('-', self.work_sheet[ProjectSheet.PROJECT_NAME_USER_SHEET].value)[0]
```

Comment:
Splitting on only one character is supported by the standard string method 'split'. So this would be:
`self.work_sheet[ProjectSheet.PROJECT_NAME_USER_SHEET].value.split('-')[0]` instead.